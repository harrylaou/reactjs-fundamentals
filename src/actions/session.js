export const LOG_OUT = 'LOG_OUT'
export const LOGGED_IN = 'LOGGED_IN'

export const logout = () => ({
  type: LOG_OUT
})

export const loggedIn = token => ({
  type: LOGGED_IN,
  token
})
