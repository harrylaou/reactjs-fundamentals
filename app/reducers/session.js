import { LOG_OUT, LOGGED_IN } from "../actions/session"
import { getSessionToken } from '../utils/localStorage'

// There is a better way to initialize the state with the store initial_state and the
// Redux middleware. We will refactor this when we explain you the Redux middleware
// Do you see any problem by doing this here? coupling? testing?
const token = getSessionToken()

const session = (state = { token }, action) => {
  switch (action.type) {
    case LOGGED_IN:
      return { ...state, token: action.token }
      break;
    case LOG_OUT:
      return { ...state, token: null }
      break;
    default:
      return state
  }
}

export default session
