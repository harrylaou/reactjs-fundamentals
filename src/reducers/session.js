import { LOG_OUT, LOGGED_IN } from "../actions/session"
import { getSessionToken } from '../utils/localStorage'

// There is a better way to initialize the state with the store initial_state and the
// Redux middleware. We will refactor this when we explain you the Redux middleware
// Do you see any problem by doing this here? coupling? testing?
const token = getSessionToken()

const session = (state = { token }, action) => {
  switch (action.type) {
    case LOGGED_IN:
      return { ...state, token: action.token }
    case LOG_OUT:
      return { ...state, token: null }
    default:
      return state
  }
}

export default session
