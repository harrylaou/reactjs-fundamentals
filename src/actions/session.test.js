import {logout, loggedIn, LOG_OUT, LOGGED_IN} from './session'

test(`Return an Action tyoe ${LOGGED_IN}`, () => {
  const token = "123456"
  expect(loggedIn(token)).toEqual({
    type: LOGGED_IN,
    token
  })
  
  expect(loggedIn(token)).toMatchSnapshot()
})

test(`Return an Action tyoe ${LOG_OUT}`, () => {
  const user = "Francis"
  expect(logout()).toEqual({
    type: LOG_OUT
  })

  expect(logout()).toMatchSnapshot()
})


