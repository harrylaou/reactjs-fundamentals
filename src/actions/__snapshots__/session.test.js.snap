// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return an Action tyoe LOG_OUT 1`] = `
Object {
  "type": "LOG_OUT",
}
`;

exports[`Return an Action tyoe LOGGED_IN 1`] = `
Object {
  "token": "123456",
  "type": "LOGGED_IN",
}
`;
